tests/trainer/test_trainer.py

```diff
@@ -1222,87 +1222,85 @@ def test_trainer_works_with_dict(self):
         train_dataset = RegressionDataset()
         eval_dataset = RegressionDataset()
         model = RegressionDictModel()
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            args = TrainingArguments(tmp_dir, report_to="none")
-            trainer = Trainer(model, args, train_dataset=train_dataset, eval_dataset=eval_dataset)
-            trainer.train()
-            _ = trainer.evaluate()
-            _ = trainer.predict(eval_dataset)
+        args = TrainingArguments(self.get_auto_remove_tmp_dir(), report_to="none")
+        trainer = Trainer(model, args, train_dataset=train_dataset, eval_dataset=eval_dataset)
+        trainer.train()
+        _ = trainer.evaluate()
+        _ = trainer.predict(eval_dataset)
 
     def test_evaluation_with_keys_to_drop(self):
         config = GPT2Config(vocab_size=100, n_positions=128, n_embd=32, n_layer=3, n_head=4)
         tiny_gpt2 = GPT2LMHeadModel(config)
         x = torch.randint(0, 100, (128,))
         eval_dataset = RepeatDataset(x)
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            args = TrainingArguments(tmp_dir, report_to="none")
-            trainer = Trainer(tiny_gpt2, args, eval_dataset=eval_dataset)
-            # By default the past_key_values are removed
-            result = trainer.predict(eval_dataset)
-            self.assertTrue(isinstance(result.predictions, np.ndarray))
-            # We can still get them by setting ignore_keys to []
-            result = trainer.predict(eval_dataset, ignore_keys=[])
-            self.assertTrue(isinstance(result.predictions, tuple))
-            self.assertEqual(len(result.predictions), 2)
+        args = TrainingArguments(self.get_auto_remove_tmp_dir(), report_to="none")
+        trainer = Trainer(tiny_gpt2, args, eval_dataset=eval_dataset)
+        # By default the past_key_values are removed
+        result = trainer.predict(eval_dataset)
+        self.assertTrue(isinstance(result.predictions, np.ndarray))
+        # We can still get them by setting ignore_keys to []
+        result = trainer.predict(eval_dataset, ignore_keys=[])
+        self.assertTrue(isinstance(result.predictions, tuple))
+        self.assertEqual(len(result.predictions), 2)
 
     def test_training_arguments_are_left_untouched(self):
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            trainer = get_regression_trainer(output_dir=tmp_dir)
-            trainer.train()
-            args = TrainingArguments(tmp_dir, report_to=[])
-            dict1, dict2 = args.to_dict(), trainer.args.to_dict()
-            for key in dict1.keys():
-                # Logging dir can be slightly different as they default to something with the time.
-                if key != "logging_dir":
-                    self.assertEqual(dict1[key], dict2[key])
+        output_dir = self.get_auto_remove_tmp_dir()
+        trainer = get_regression_trainer(output_dir=output_dir)
```

Comment:
I probably would name it as `tmp_dir` but it's just my own taste. No need to change unless if you feel the same.